### ISSUE TITLE
Writing sbt.version vs scripted filtering

### ISSUE THREAD

**Issue by dwijnand** [sbt/sbt#4536]
## steps

1. Write a scripted test
1. Run sbt in that scripted test

## problem

sbt will write out `sbt.version` under `project/build.properties` which acts as a test filter.

## expectation

The, possibly discouraged, practice of running sbt in a scripted test shouldn't implicitly alter the test setup.

## notes
sbt version: 1.2.8

Possible solutions:
One possible solution is just before writing the file sbt can detect it's running a scripted test and skip writing.  Maybe a stupid "Am I inside ../../sbt-test?" 😕

Or maybe the filtering shouldn't be based on `sbt.version` in `project/build.properties` but on something else (let's say, temporarily, `sbt.filter.version`).  In addition, the filter version could be a bit more flexible than a flat version, like maybe `^1.1`, `~1.2`, or `>=1.0.0, <1.1.0`.  See:

* https://docs.npmjs.com/misc/semver#caret-ranges-123-025-004
* https://docs.npmjs.com/misc/semver#tilde-ranges-123-12-1
* https://docs.npmjs.com/misc/semver#ranges

This solution I'd say should include giving a deprecation warning about the old use of sbt.version for filtering, and maybe that warning could also educate the sbt user that you can temporarily change the sbt version with the system property sbt.version: `sbt -Dsbt.version=1.2.1`.

**Comment by dwijnand** [comment]
(Ah, the eternal dilemma.  Is a feature interaction a bug fix or an enhancement...?)

**Comment by olafurpg** [comment]
Would it be possible to prompt the user whether to create build.properties?

```
No sbt build detected in directory $PATH. Would you like to create one? (y/N): 
```

I have several times accidentally started sbt in the wrong subdirectories (non-scripted) of my workspace and I have to delete the created `build.properties`.

**Comment by dwijnand** [comment]
Yeah, maybe.  I say maybe because we've had some problems in the past with trying to inform/interact with the user and also trying to detect interactive mode.

But that chance would probably be here: https://github.com/sbt/sbt/blob/v1.2.8/main/src/main/scala/sbt/Main.scala#L914

**Comment by eed3si9n** [comment]
> 2\. Run sbt in that scripted test

When you say "run in that scripted test" do you  mean running in the directory, without first copying it to `/tmp`?

**Comment by dwijnand** [comment]
That's right, people do that.  I'm fairly sure I've done it too, and had to clean up sbt's littering.

**Comment by eed3si9n** [comment]
Then sbt works as currently designed, so I don't think this is a bug (let's define this as _unintended behavior_), and I also disagree with the expectation. It's a minor inconvenience of an otherwise good behavior, and I think we should keep it stupid.

**Comment by dwijnand** [comment]
I think you'd have a leg to stand on if it were documented both that:
* running sbt in a scripted test is (strongly?) discouraged, and that
* setting sbt.version in scripted tests filters.
